SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialShadowActor.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+#include "SpatialConstants.h"
+#include "SpatialShadowActor.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialShadowActor, Log, All);
+
+UCLASS(Transient)
+class SPATIALGDK_API USpatialShadowActor : public UObject
+{
+	GENERATED_BODY()
+
+public:
+	USpatialShadowActor(const FObjectInitializer& ObjectInitializer = FObjectInitializer::Get());
+
+	void Init(const Worker_EntityId InEntityId, AActor* InActor);
+
+	void CheckUnauthorisedDataChanges(const Worker_EntityId InEntityId, AActor* InActor);
+
+	void Update(const Worker_EntityId InEntityId, AActor* InActor);
+
+private:
+	Worker_EntityId EntityId;
+	AActor* Actor;
+	FString ReplicatedPropertyHash;
```

Comment:
Could/should this be an integer hash rather than a string, assuming they only need to be generated and compared?

Reply:
+1

Reply:
Yes replaced with an int hash.